In [1]:
import os
import sys
sys.path.append("../src")
from utils.common.pathManager import FilePath
from collections import defaultdict
path = FilePath("/media/mrtc101/Martín SSD/tesis/datasets/xBD")
print(path.get_folder_names())

['hold', 'Selected imgs', 'test', 'tier3', 'train']


In [2]:
count_types = defaultdict(lambda : 0)
for folder_path in path.get_folder_paths():
    for split_name in folder_path.get_folder_names():
        if(split_name == "images"):
            split_path = folder_path.join(split_name)
            for file in split_path.get_files_names():
                if file.find("pre") > -1:
                    prefix = file.split("_")[0]
                    count_types[prefix] += 1
[(t,n) for t,n in count_types.items()]

[('guatemala-volcano', 28),
 ('hurricane-florence', 546),
 ('midwest-flooding', 445),
 ('socal-fire', 1403),
 ('hurricane-matthew', 405),
 ('hurricane-harvey', 522),
 ('hurricane-michael', 550),
 ('santa-rosa-wildfire', 377),
 ('palu-tsunami', 196),
 ('mexico-earthquake', 193),
 ('joplin-tornado', 149),
 ('lower-puna-volcano', 291),
 ('moore-tornado', 227),
 ('nepal-flooding', 619),
 ('pinery-bushfire', 1845),
 ('portugal-wildfire', 1869),
 ('sunda-tsunami', 148),
 ('tuscaloosa-tornado', 343),
 ('woolsey-fire', 878)]

('mexico-earthquake', 193)

('nepal-flooding', 619)
('midwest-flooding', 445)

('santa-rosa-wildfire', 377)
('portugal-wildfire', 1869)
('socal-fire', 1403)
('woolsey-fire', 878)
('pinery-bushfire', 1845)

('palu-tsunami', 196)
('sunda-tsunami', 148)

('joplin-tornado', 149)
('moore-tornado', 227)
('tuscaloosa-tornado', 343)

('hurricane-florence', 546)
('hurricane-matthew', 405)
('hurricane-harvey', 522)
('hurricane-michael', 550)

('guatemala-volcano', 28)
('lower-puna-volcano', 291)


In [3]:
[t for t,n in count_types.items()]

['guatemala-volcano',
 'hurricane-florence',
 'midwest-flooding',
 'socal-fire',
 'hurricane-matthew',
 'hurricane-harvey',
 'hurricane-michael',
 'santa-rosa-wildfire',
 'palu-tsunami',
 'mexico-earthquake',
 'joplin-tornado',
 'lower-puna-volcano',
 'moore-tornado',
 'nepal-flooding',
 'pinery-bushfire',
 'portugal-wildfire',
 'sunda-tsunami',
 'tuscaloosa-tornado',
 'woolsey-fire']